crates/but-testsupport/src/lib.rs

```diff
@@ -304,20 +306,22 @@ pub fn visualize_tree(tree_id: gix::Id<'_>) -> termtree::Tree<String> {
 pub fn visualize_disk_tree_skip_dot_git(root: &Path) -> anyhow::Result<termtree::Tree<String>> {
     use std::os::unix::fs::MetadataExt;
     fn normalize_mode(mode: u32) -> u32 {
-        match mode {
-            0o40777 => 0o40755,
-            0o40775 => 0o40755,
-            0o10664 => 0o10644,
-            0o10666 => 0o10644,
-            0o100664 => 0o100644,
-            0o100666 => 0o100644,
-            0o100775 => 0o100755,
-            0o100777 => 0o100755,
-            0o120775 => 0o120755,
-            0o120777 => 0o120755,
-            other => other,
+        let normalize_permission_bits = |mode: u32| if mode & 0o100 == 0 { 0o644 } else { 0o755 };
+
+        match mode & FILETYPE_MASK {
+            // File types Git cares about
+            directory @ 0o40000 => directory | normalize_permission_bits(mode),
+            symlink @ 0o120000 => symlink | normalize_permission_bits(mode),
```

Comment:
From a Git perspective it's a bit weird to propagate permission bits on directories and symlinks, as Git does not store those. But the current snapshots do so.

Do we want to "properly" normalize this to a more Git-aligned view (i.e. zero out the permission bits for these types in existing snapshots), or is this fine?

Reply:
Yes, why not go with the minimal version that also can't panic. Let's merge this as well and you can follow it up.